### PR TITLE
feat: add integration test for preview

### PR DESCRIPTION
**Pull request by mr25mr**
This PR 
* Add a new package `“@sap-ux-private/playwright”`
* Add simple integration test for preview. It test preview with different UI5 Versions:
    * 1.84.35
    * 1.96.22
    * 1.108.19
    * 1.111.8
    * 1.114.0
    * 1.115.0
* Add Github Actions
* Pave the ground for integration test for CPE on adaption project. Integration tests for CPE will follow in another PR.

### CODE REVIEW

File: .github/workflows/pipeline.yml
Comment (devinea) [inline .github/workflows/pipeline.yml:68]: @mr25mr Have you tried to use NX caching instead of detecting changes?
See test for example:
https://github.com/SAP/open-ux-tools/blob/main/nx.json#L9
https://github.com/SAP/open-ux-tools/blob/main/nx.json#L42-L52

Also with node_module caching we can probably skip `npx playwright install --with-deps`
https://github.com/SAP/open-ux-tools/blob/main/.github/workflows/pipeline.yml#L26-L34

Comment (mr25mr) [reply .github/workflows/pipeline.yml:68]: @devinea I tried to use NX caching, does not seems to be straightforward to achieve same goal. We need to have playwright reports for each os with combination of node version e.g `playwright-reports-os-windows-latest-node-version-18.x` and using NX caching for this goal seemed a bit complicated.

regarding `npx playwright install --with-deps` we need to install browser dependencies due to this issue https://github.com/SAP/open-ux-tools/pull/1507/files#diff-0fc4e4b47236265ffb9c3ec977c54445ee6e60856bd910f44232abe7a91e1316R41
Here is official doc. https://playwright.dev/docs/ci-intro#setting-up-github-actions

one other constrains - [Each version of Playwright needs specific versions of browser binaries to operate. You will need to use the Playwright CLI to install these browsers](https://playwright.dev/docs/browsers#introduction). Caching them, not sure.

File: packages/preview-middleware/test/integration/preview/preview.spec.ts
Comment (zdravko-georgiev) [inline packages/preview-middleware/test/integration/preview/preview.spec.ts:81]: Why do you add this? We want to test against the `preview-middleware`. The other middlewares should be installed.Right?

File: .eslintrc
Comment (zdravko-georgiev) [inline .eslintrc:110]: Why are you disabling those?

File: .changeset/bright-vans-add.md
Comment (zdravko-georgiev) [inline .changeset/bright-vans-add.md:5]: I am not sure, if we actually need a changeset. We are not changing anything in the functionality of the package

File: packages/playwright/package.json
Comment (mr25mr) [reply packages/playwright/package.json:2]: Yes, currently there is no such request and changing it to public should be easy once there is such request. I also do not have any objection to make it public now, but [admin](https://github.com/orgs/SAP/teams/ux-tools-admin) should agree

File: packages/preview-middleware/test/integration/preview/preview.spec.ts
Comment (mr25mr) [reply packages/preview-middleware/test/integration/preview/preview.spec.ts:81]: yes, adding as project dependencies mean we test agains release and adding like this actually test current main code base. There might not be much difference code wise, but testing current code base detect regression if any.

File: .changeset/bright-vans-add.md
Comment (mr25mr) [reply .changeset/bright-vans-add.md:5]: in UI5 LA if there is any changes in any package, we include them in change set. I think we do same strategies in open-ux tools. Removing it  may cause pipeline checks fails.

File: packages/playwright/package.json
Comment (zdravko-georgiev) [reply packages/playwright/package.json:2]: @SAP/ux-tools-admin Any objections to make this public?

File: packages/preview-middleware/test/integration/preview/preview.spec.ts
Comment (zdravko-georgiev) [reply packages/preview-middleware/test/integration/preview/preview.spec.ts:81]: OK, then please remove the `backend-proxy-middleware`. Once this is submitted we can introduce a similar integration test for the `backend-proxy-middleware`.

File: .eslintrc
Comment (zdravko-georgiev) [reply .eslintrc:110]: Ah, OK. Got it, thanks.

File: .changeset/bright-vans-add.md
Comment (zdravko-georgiev) [reply .changeset/bright-vans-add.md:5]: I think that we can safely remove the changeset in this case. @devinea Is my assumption correct?

File: .github/workflows/pipeline.yml
Comment (devinea) [reply .github/workflows/pipeline.yml:68]: As discussed https://github.com/SAP/open-ux-tools/pull/1517/commits/e3d0dcdae7a00e1af47a3906b9359496b7f3c66c seems like a viable approach using NX.
Playwright installed is not cached with that approach. But something like this looks nice https://github.com/microsoft/playwright/issues/7249#issuecomment-1385567519

File: packages/playwright/package.json
Comment (devinea) [reply packages/playwright/package.json:2]: I am not opposed to making it public. 
But private is ok for now it we don't have concrete plans to consume back in tools-suite. It can be changed later.

File: .changeset/bright-vans-add.md
Comment (devinea) [reply .changeset/bright-vans-add.md:5]: If the new module is private then a changeset is not needed.

Comment (devinea) [reply .changeset/bright-vans-add.md:5]: On second thoughts, to write the change log properly it is best to include the changeset. 
Publishing to npm is not applicable in this case.

Comment (zdravko-georgiev) [reply .changeset/bright-vans-add.md:5]: OK

File: .github/workflows/pipeline.yml
Comment (mr25mr) [reply .github/workflows/pipeline.yml]: @devinea address in this commit https://github.com/SAP/open-ux-tools/pull/1507/commits/c889140a1b5509489577f01165fe1cf09e93dae3
Currently playwright is installed in mono repo only once, and when building [key for caching  ](https://github.com/SAP/open-ux-tools/pull/1507/files#diff-fe700949ba86403864aa7b6fdcab2c00c55e1047f73a6a282135a35e8a21e03bR54) we do not consider playwright version. It should be ok as long as we do not use different playwright versions. Getting playwright version to include it in key, does not work reliable on windows at least I tried a couple of time :)

**Review by donal-tobin-sap — DISMISSED** [review]
Looks to be ok from a modules consumption point of view.

File: .eslintrc
Comment (tobiasqueck) [reply .eslintrc:110]: Aehm, no. Why do you need to disable those for your tests?

**Review by devinea — DISMISSED** [review]
New module setup and pipeline setup looks good. 
Changeset ✅ 
Private module ✅ 
Thanks for the updates.

File: .eslintrc
Comment (mr25mr) [reply .eslintrc:110]: @tobiasqueck since this is only test file, we will work on those in a different PR after alignment on what rules should be switched off and which rules should not. I hope this is fine with you. Thanks.

Comment (tobiasqueck) [reply .eslintrc:110]: I am sorry, but no. If you need the exception for a specific project temporarily, then put them there, but do not put them as global rules in the root folder.

**Review by zdravko-georgiev — APPROVED** [review]
Changes to preview-middleware look good
Changeset is OK
Review comments addressed

One thing that is missing is assigning code owners for the new package. But this can be done also separately.

**Comment by tobiasqueck** [comment]
No objections from my side anymore, as soon as you have your 2nd approval, you can merge.

**Review by devinea — APPROVED** [review]
Reapproving

File: pnpm-lock.yaml
Comment (mr25mr) [reply pnpm-lock.yaml:9932]: @donal-tobin-sap  the `axios` module is not added in new module or to existing module which I add integration test to it. This change in pnpm-lock.yaml might have been due to result of merging with main branch.

**Pull request merged**
